src/x86/avx2.rs

```diff
@@ -1,3 +1,551 @@
+pub fn _mm256_cmpgt_epi16(a: i16x16, b: i16x16) -> i16x16 {
+    a.gt(b)
+}
+
+/// Compare packed 8-bit integers in `a` and `b` for greater-than.
+#[inline(always)]
+#[target_feature = "+avx2"]
+pub fn _mm256_cmpgt_epi8(a: i8x32, b: i8x32) -> i8x32 {
+    a.gt(b)
+}
+
+#[allow(improper_ctypes)]
+extern "C" {
+    #[link_name = "llvm.x86.avx2.pabs.b"]
+    fn pabsb(a: i8x32) -> i8x32;
+    #[link_name = "llvm.x86.avx2.pabs.w"]
+    fn pabsw(a: i16x16) -> i16x16;
+    #[link_name = "llvm.x86.avx2.pabs.d"]
+    fn pabsd(a: i32x8) -> i32x8;    
+    #[link_name = "llvm.x86.avx2.padds.b"]
+    fn paddsb(a: i8x32, b: i8x32) -> i8x32;
+    #[link_name = "llvm.x86.avx2.padds.w"]
+    fn paddsw(a: i16x16, b: i16x16) -> i16x16;
+    #[link_name = "llvm.x86.avx2.paddus.b"]
+    fn paddusb(a: u8x32, b: u8x32) -> u8x32;
+    #[link_name = "llvm.x86.avx2.paddus.w"]
+    fn paddusw(a: u16x16, b: u16x16) -> u16x16;
+    #[link_name = "llvm.x86.avx2.pavg.b"]
+    fn pavgb(a: u8x32, b: u8x32) -> u8x32;
+    #[link_name = "llvm.x86.avx2.pavg.w"]
+    fn pavgw(a: u16x16, b: u16x16) -> u16x16;
+    #[link_name = "llvm.x86.avx2.pblendvb"]
+    fn pblendvb(a: i8x32, b: i8x32, mask: __m256i) -> i8x32;
+}
+
+
+#[cfg(test)]
+mod tests {
+    use v256::*;
+    use x86::avx2;
+    use x86::__m256i;
+    use std;
+
+    #[test]
+    #[target_feature = "+avx2"]
+    fn _mm256_abs_epi32() {
+        let a = i32x8::new(0, 1, -1, std::i32::MAX, 
+                           std::i32::MIN + 1, 100, -100, -32);
+        let r = avx2::_mm256_abs_epi32(a);
+        let e = i32x8::new(0, 1, 1, std::i32::MAX, 
+                           (std::i32::MIN + 1).abs(), 100, 100, 32);
+        assert_eq!(r, e);
+    }
+
+    #[test]
+    #[target_feature = "+avx2"]
+    fn _mm256_abs_epi16() {
+        let a = i16x16::new(0, 1, -1, 2, 
+                            -2, 3, -3, 4, 
+                            -4, 5, -5, std::i16::MAX, 
+                            std::i16::MIN + 1, 100, -100, -32);
+        let r = avx2::_mm256_abs_epi16(a);
+        let e = i16x16::new(0, 1, 1, 2, 
+                            2, 3, 3, 4, 
+                            4, 5, 5, std::i16::MAX, 
+                            (std::i16::MIN + 1).abs(), 100, 100, 32);
+        assert_eq!(r, e);
+    }
+
+    #[test]
+    #[target_feature = "+avx2"]
+    fn _mm256_abs_epi8() {
+        let a = i8x32::new(0, 1, -1, 2, 
+                           -2, 3, -3, 4, 
+                           -4, 5, -5, std::i8::MAX, 
+                           std::i8::MIN + 1, 100, -100, -32, 
+                           0, 1, -1, 2,
+                           -2, 3, -3, 4,
+                           -4, 5, -5, std::i8::MAX, 
+                           std::i8::MIN + 1, 100, -100, -32);
+        let r = avx2::_mm256_abs_epi8(a);
+        let e = i8x32::new(0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, std::i8::MAX, (std::i8::MIN + 1).abs(), 100, 100, 32, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, std::i8::MAX, (std::i8::MIN + 1).abs(), 100, 100, 32);
```

Comment:
Thanks. :-) And thank you for working on this! It's a huge help. There are *a lot* of intrinsics to work through. :-)